pandas/_libs/lib.pyx

```diff
@@ -647,7 +646,7 @@ def row_bool_subset(ndarray[float64_t, ndim=2] values,
                 out[pos, j] = values[i, j]
             pos += 1
 
-    return out
+    return out.base  # `.base` to access underlying np.ndarray
```

Comment:
I know elsewhere we do `np.asarray(<memoryview>)` - are those equivalent?

Reply:
Very close to equivalent.  It's a bit roundabout.

Above `out` is typed as a `float64_t[:, :]` memoryview, but then instantiated with 

```
out = np.empty((mask.sum(), k), dtype=np.float64)
```

In executing that line, cython creates the numpy array, then casts it as a memoryview `out`, and attaches the original ndarray as `out.base`.  So `out.base` is an attribute lookup to return an existing np.ndarray.

The performance penalty to `np.asarray` must be tiny; it might be worth it to do it that way to avoid confusion.

Reply:
ahh yes, can we change these (separate PR)